dask_geopandas/io/arrow.py

```diff
@@ -0,0 +1,212 @@
+
+
+if TYPE_CHECKING:
+    import pyarrow
+
+
+class ArrowDatasetEngine:
+
+    file_format: str
+
+    @classmethod
+    def read_metadata(cls, fs, paths, columns, filters, index):
+
+        import pyarrow.dataset as ds
+        from pyarrow.parquet import _filters_to_expression
+
+        # dataset discovery
+        # TODO support filesystems
+        if len(paths) == 1:
+            # list of 1 directory path is not supported
+            paths = paths[0]
+        dataset = ds.dataset(
+            paths, partitioning="hive", filesystem=None, format=cls.file_format
+        )
+
+        # Get all (filtered) fragments
+        if filters is not None:
+            filter = _filters_to_expression(filters)
+        else:
+            filter = None
+
+        fragments = list(dataset.get_fragments(filter=filter))
+
+        # numeric rather than glob ordering
+        # TODO how does this handle different partitioned directories?
+        fragments = sorted(fragments, key=lambda f: natural_sort_key(f.path))
+
+        # TODO potential splitting / aggregating of fragments
+
+        # Create dask meta
+        schema = dataset.schema
+        # TODO add support for `categories`keyword
+        dtypes = _get_pyarrow_dtypes(schema, categories=None)
+        if columns is not None:
+            ex = set(columns) - set(dtypes)
+            if ex:
+                raise ValueError(
+                    "Requested columns (%s) not in schema (%s)" % (ex, set(dtypes))
+                )
+        columns = list(dtypes) if columns is None else columns
+        index = [index] if isinstance(index, str) else index
+        meta = _meta_from_dtypes(columns, dtypes, index, [])
+        return fragments, meta, schema, filter
+
+    @classmethod
+    def _arrow_table_to_pandas(
+        cls, arrow_table: "pyarrow.Table", categories, **kwargs
+    ) -> pd.DataFrame:
+        _kwargs = kwargs.get("arrow_to_pandas", {})
+        _kwargs.update({"use_threads": False, "ignore_metadata": False})
+
+        return arrow_table.to_pandas(categories=categories, **_kwargs)
+
+    @classmethod
+    def read_partition(cls, fs, fragment, schema, columns, filter, **kwargs):
+
+        table = fragment.to_table(
+            schema=schema, columns=columns, filter=filter, use_threads=False
+        )
+        df = cls._arrow_table_to_pandas(table, None)
+        return df
+
+
+class GeoDatasetEngine:
+    @classmethod
+    def _arrow_table_to_pandas(
+        cls, arrow_table: "pyarrow.Table", categories, **kwargs
+    ) -> pd.DataFrame:
+        from geopandas.io.arrow import _arrow_to_geopandas
+
+        _kwargs = kwargs.get("arrow_to_pandas", {})
+        _kwargs.update({"use_threads": False, "ignore_metadata": False})
+
+        # TODO support additional keywords
+        try:
+            return _arrow_to_geopandas(arrow_table)
+        except ValueError as err:
+            # when no geometry column is selected, the above will error.
```

Comment:
To do more robust fallback to read the file even when no geometry columns are included in the user-defined columns to read from the file, you need to compare the user-defined columns against the geo columns listed in the metadata prior to `_arrow_to_geopandas`.

Alternatively, in geopandas we could consider adding an optional keyword to ignore this error gracefully instead of raising an exception (e.g., `missing_geometry="ignore" or something like that).

But I'm not clear on what the advantage is here of supporting reads of geo feather files and specifically ignoring their geometry columns.

Reply:
> But I'm not clear on what the advantage is here of supporting reads of geo feather files and specifically ignoring their geometry columns.

This is specifically to support that you do `ddf = dask_geopandas.read_feather(...); ddf["attribute"]....compute()`, where `ddf` is a GeoDataFrame but the result will be a normal DataFrame, because of pushing down the column selection.

But as you say, this could also be done by changing geopandas to not error when no geometry column is present. But so with current geopandas, the above is needed.